src/Esprima/Loc.cs

```diff
@@ -1,29 +1,52 @@
-﻿using System.Diagnostics;
-using System.Globalization;
+﻿using System.Globalization;
+using System.Runtime.CompilerServices;
 
 namespace Esprima;
 
-[DebuggerDisplay("{ToString()}")]
 public readonly struct Location : IEquatable<Location>
 {
     public readonly Position Start;
     public readonly Position End;
     public readonly string? Source;
 
-    public Location(in Position start, in Position end) : this(start, end, null)
+    private static bool Validate(in Position start, in Position end, bool @throw = true)
```

Comment:
instead of reserved word, `throwOnError` would probably be better (like in `Type.GetType`)